test/repositories/EmailRepositorySpec.scala

```diff
@@ -70,64 +74,24 @@ class EmailRepositorySpec extends SpecBase {
     })
   }
 
-  "return 'false' if no update performed" in {
-    val eori = "UnknownEori"
-    val otherEori = "someEori"
+  "return 'false' if no update performed" in new Setup {
+    val unknownEori = "UnknownEori"
     val notificationEmail = NotificationEmail("some@email.com", DateTime.now(), None)
 
-    val undeliverableInformationEvent: UndeliverableInformationEvent = UndeliverableInformationEvent(
-      "some-id",
-      "some event",
-      "some@email.com",
-      "detected",
-      Some(12),
-      Some("unknown reason"),
-      s"HMRC-CUS-ORG~EORINumber~$eori",
-      Some("sdds")
-    )
-
-    val undeliverableInformation: UndeliverableInformation =
-      UndeliverableInformation(
-        "some-subject",
-        "some-event-id",
-        "some-group-id",
-        DateTime.now(),
-        undeliverableInformationEvent
-      )
     await(for {
```

Comment:
Please remove await and update as below

    for {
      _ <- repository.set(eori, notificationEmail)
      result <- repository.findAndUpdate(unknownEori, undeliverableInformation)
      _ = result.map(notif => notif mustBe None)
      record <- repository.get(unknownEori)
      _ <- dropData()
    } yield {
      record mustBe None
    }